tests/scurl.sh

```diff
@@ -44,15 +44,18 @@ date=$(date "+%a, %d %b %Y %H:%M:%S %Z")
 
 req_digest=$(echo -n "$request" | openssl dgst -sha256 -binary | openssl base64)
 
+content_length=${#request}
+
 # Construct string to sign
 string_to_sign="date: $date
-digest: SHA-256=$req_digest"
+digest: SHA-256=$req_digest
+content-length: $content_length"
```

Comment:
Why do we need to do this? I would have thought curl would set the content length?

Reply:
Curl does set the content-length, but we need to sign it. If its unsigned we remove it during canonicalisation, and then if anyone else tries to parse the raw canonicalised form, they have no `Content-Length` and `http_parser` treats this as an empty request.

Reply:
Ah I see, of course.